src/neo/Plugins/Plugin.cs

```diff
@@ -58,74 +59,108 @@ protected Plugin()
             Configure();
         }
 
-        public abstract void Configure();
-
-        protected virtual void OnPluginsLoaded()
+        protected virtual void Configure()
         {
         }
 
         private static void ConfigWatcher_Changed(object sender, FileSystemEventArgs e)
         {
-            foreach (var plugin in Plugins)
+            switch (GetExtension(e.Name))
             {
-                if (plugin.ConfigFile == e.FullPath)
-                {
-                    plugin.Configure();
-                    plugin.Log($"Reloaded config for {plugin.Name}");
+                case ".json":
+                    Plugins.FirstOrDefault(p => p.ConfigFile == e.FullPath)?.Configure();
                     break;
-                }
+                case ".dll":
+                    if (e.ChangeType != WatcherChangeTypes.Created) return;
+                    if (GetDirectoryName(e.FullPath) != PluginsDirectory) return;
+                    LoadPlugin(Assembly.Load(File.ReadAllBytes(e.FullPath)));
```

Comment:
Auto load plugins maybe is not a good idea

Reply:
Why?

Reply:
Because you don't know if it's a user action, it could come from the install command, or because someone just copy this library when the node is running. I understand that then should be a rights problem, because you shouldn't be able to copy to that folder. But it's needed to open that door?

Reply:
If someone copied a dll to the plugin directory, even if you don't load it dynamically, it will be loaded the next time the cli starts. So there is actually no difference.